rmf_fleet_adapter/src/rmf_fleet_adapter/agv/FleetUpdateHandle.cpp

```diff
@@ -61,217 +72,564 @@ class LiaisonNegotiator : public rmf_traffic::schedule::Negotiator
+        name.c_str(), delivery.pickup_place_name.c_str(), id.c_str());
+
+        return;
+    }
+
+    const auto dropoff_wp = graph.find_waypoint(delivery.dropoff_place_name);
+    if (!dropoff_wp)
+    {
+      RCLCPP_INFO(
+        node->get_logger(),
+        "Fleet [%s] does not have a named waypoint [%s] configured in its "
+        "nav graph. Rejecting BidNotice with task_id:[%s]",
+        name.c_str(), delivery.dropoff_place_name.c_str(), id.c_str());
+
+        return;
+    }
+
+    new_request = rmf_task::requests::Delivery::make(
+      id,
+      pickup_wp->index(),
+      delivery.pickup_dispenser,
+      dropoff_wp->index(),
+      delivery.dropoff_ingestor,
+      delivery.items,
+      motion_sink,
+      ambient_sink,
+      planner,
+      start_time,
+      drain_battery);
+
+    RCLCPP_INFO(
+      node->get_logger(),
+      "Generated Delivery request");
+
+  }
+  else if (task_type.type == rmf_task_msgs::msg::TaskType::TYPE_LOOP)
+  {
+    const auto& loop = task_profile.loop;
+    if (loop.start_name.empty())
+    {
+      RCLCPP_INFO(
+        node->get_logger(),
+        "Required param [loop.start_name] missing in TaskProfile."
+        "Rejecting BidNotice with task_id:[%s]" , id.c_str());
+
+      return;
+    }
+
+    if (loop.finish_name.empty())
+    {
+      RCLCPP_INFO(
+        node->get_logger(),
+        "Required param [loop.finish_name] missing in TaskProfile."
+        "Rejecting BidNotice with task_id:[%s]" , id.c_str());
+
+      return;
+    }
+
+    if (loop.num_loops < 1)
+    {
+      RCLCPP_INFO(
+        node->get_logger(),
+        "Required param [loop.num_loops] in TaskProfile is invalid."
+        "Rejecting BidNotice with task_id:[%s]" , id.c_str());
+
+      return;
+    }
+
+    const auto start_wp = graph.find_waypoint(loop.start_name);
+    if (!start_wp)
+    {
+      RCLCPP_INFO(
+        node->get_logger(),
+        "Fleet [%s] does not have a named waypoint [%s] configured in its "
+        "nav graph. Rejecting BidNotice with task_id:[%s]",
+        name.c_str(), loop.start_name.c_str(), id.c_str());
+
+        return;
+    }
+
+    const auto finish_wp = graph.find_waypoint(loop.finish_name);
+    if (!finish_wp)
+    {
+      RCLCPP_INFO(
+        node->get_logger(),
+        "Fleet [%s] does not have a named waypoint [%s] configured in its "
+        "nav graph. Rejecting BidNotice with task_id:[%s]",
+        name.c_str(), loop.finish_name.c_str(), id.c_str());
+
+        return;
+    }
+
+    new_request = rmf_task::requests::Loop::make(
+      id,
+      start_wp->index(),
+      finish_wp->index(),
+      loop.num_loops,
+      motion_sink,
+      ambient_sink,
+      planner,
+      start_time,
+      drain_battery);
+
+    RCLCPP_INFO(
+      node->get_logger(),
+      "Generated Loop request");
+  }
+  else
+  {
+    RCLCPP_INFO(
+      node->get_logger(),
+      "Invalid TaskType in TaskProfile. Rejecting BidNotice with task_id:[%s]",
```

Comment:
I'd suggest including the `task_type.type` information in this message and escalate it to `_WARN` or `_ERROR`.

Reply:
Done https://github.com/osrf/rmf_core/pull/217/commits/f73f6896aedf9ba33247fe43c9034138912a848a